### ISSUE TITLE
Write API with Master Token creates express session

### ISSUE THREAD

**Issue by uplift** [NodeBB/NodeBB#9450]
Recently we completed a move from Themes to API/Sockets integration on our website. We are currently on an old version of Nodebb (1.10.2 with a plan to upgrade soon) with the Write API plugin being used for API access. The Write API plugin uses a Master Token and _uid to get user data. One thing we've noticed is a spike in the number of express sessions stored in our Redis Db as a result of this approach. It seems with every API call using this method it creates an express session which is then never used again and is only cleared down after the TTL expires.

Is there a way of disabling express session creation for these types of API calls? They don't serve any purpose as far as I can tell as once the response is received the session is done and it is increasing the size of our db instance.

Not sure if this is the same with the new Core Write API?

**Comment by julianlam** [comment]
Hi @uplift -- that's an interesting report, I'll take a look and see if maybe we can stop sessions from being generated on write API calls.

**Comment by julianlam** [comment]
It seems that I tried fixing this earlier by passing in `saveUnitialized` into the passport options, but this effectively took no effect because I save data into `req.session`, which by nature means a session object must be created in the session store.

I've added code to clean it up at the end of the call, if you are making an API call with a token. The v3 Write API is also called via cookie auth, but this change above should not affect it.
